[masterlist.yaml] SKSE Version checks & Messages

MacSplody opened the pull request:
- SKSE64 & SSE Exe version checks
- Compatibility check SkyrimSE.exe & skse64_loader.exe
- Message options for (RequiresSKSE, Obsolete, patched, redundant, conflicting & patch available) Mods.
- Warning message for missing SKSE loader when .dll requires it. (Could add catch all to look for any .dll in skse\plugin folder)

MacSplody committed:
Added Version Checks

- SKSE64 & SSE Exe version checks
- Compatibility check SkyrimSE.exe & skse64_loader.exe
- Message options for (RequiresSKSE, Obsolete, patched, redundant, conflicting & patch available) Mods.

MacSplody committed:
Delete .bak

MacSplody committed:
AchievementsModsEnabler Dll name corrected

MacSplody commented:
One open issue is the SkyrimSE.exe version is currently listed as 1.0.0.0 in the exe details.

![image](https://user-images.githubusercontent.com/1944639/38240298-27ebe690-3727-11e8-9dd2-d92e54bc7802.png)

Ortham commented on masterlist.yaml:
```diff
@@ -112,9 +238,69 @@ globals:
     subs: [ 'https://loot.github.io/latest-thread/' ]
+# SKYRIM SE Latest
+  - type: say
+    condition: 'file("../SkyrimSE.exe") and version("../SkyrimSE.exe", "1.0.0.0", <)'
```
Given that the executable version doesn't reflect the game version, I think it's pointless checking it. You could check the CRC instead, but I don't know if different regions get different executables (which would all have different CRCs)...

Ortham commented on masterlist.yaml:
```diff
@@ -112,9 +238,69 @@ globals:
+      - lang: ja
+        text: 'SKSE64は最新版です。'
+  - type: error
+    condition: 'version("../skse64_loader.exe", "0.2.0.6", ==) and version("../SkyrimSE.exe", "1.0.0.0", !=)'
```
Same as above, if the version is going to stay at 1.0.0.0, this isn't a terribly helpful message.

MacSplody replied:
Yeah only noticed it late in testing, unusual as FO4 and Oblivion both display the correct version.

Ortham commented on masterlist.yaml:
```diff
@@ -112,9 +238,69 @@ globals:
+# SKSE64 plugins requirement
+  - <<: *requireSKSE64
+    subs: [ 'Achievements Mods Enabler' ]
+    # url: [ 'https://www.nexusmods.com/skyrimspecialedition/mods/245' ]
```
A more useful way of encoding the link would be to include it in the message substitution, e.g. `subs: [ '[Achievements Mods Enabler](https://www.nexusmods.com/skyrimspecialedition/mods/245)' ]`. That way it'll show up as a hyperlink in the UI and won't look slightly confusing - initially I wondered why you'd left invalid syntax commented out.

MacSplody replied:
Can't claim credit for the code, it was copied from FO4 masterlist. Thanks for the tip.

Ortham commented on masterlist.yaml:
```diff
@@ -19,8 +97,56 @@ common:
+        text: "Du ser ud til at bruge %1%, men du bruger ikke “%1%”-lappen for dette plugin: %2%"
+      - lang: ja
+        text: "%1%が使用されていますが、このプラグイン用の%1%のパッチが有効になっていません: %2%"
+  - &requireSKSE64
```
As far as I can see, this is the only `common` entry you've added that's actually used, why all the others?

MacSplody replied:
Haven't used YAML before so was mainly doing it to get an idea of how it was working.
And also because those entries were in FO4 masterlist which I assumed was more active and up to date.

Ortham replied:
OK, well the FO4 masterlist is the most recent, but if an entry in the `common` section isn't referenced by its `&anchor` anywhere, it's not used and so is just padding the masterlist with noise, so it's better for it not to be there.

Ortham reviewed (changes requested):
Thanks for this, I've left my comments inline. It's unfortunate that SkyrimSE.exe's file version doesn't reflect its real version.

Ortham commented:
If you remove the unused `common` entries and the messages that have a `SkyrimSE.exe` version check, I'll be happy to merge this.

MacSplody committed:
Review Revisions

- Remove unused common entries.
- Remove SkyrimSE.exe version & SKSE compatibility check. (Unable to get true exe version at this time).
- Updated requireSKSE64 entry for Achievment mod with suggested change to link.

MacSplody commented:
- Remove unused common entries.
- Remove SkyrimSE.exe version & SKSE compatibility check. (Unable to get true exe version at this time).
- Updated requireSKSE64 entry for Achievment mod with suggested change to link.

Ortham commented on masterlist.yaml:
```diff
@@ -112,9 +131,49 @@ globals:
     subs: [ 'https://loot.github.io/latest-thread/' ]
+# SKSE Latest
+  - type: say
+    condition: 'file("../skse64_loader.exe") and version("../skse64_loader.exe", "0.2.0.6", <)'
```
I didn't think of this before, but this condition can be shortened to just `version("../skse64_loader.exe", "0.2.0.6", <)`, if the file doesn't exist it will be false.

MacSplody replied:
I'll test it in a bit and push change.

MacSplody replied:
Tested &  
- using either != and < would always show skse latest version message.
condition: 'version("../skse64_loader.exe", "0.2.0.6", !=)'
condition: 'version("../skse64_loader.exe", "0.2.0.6", <)'

Ortham replied:
Ah, sorry, my mistake, `version()` will return true if the file doesn't exist for `!=`, `<` and `<=`, though thinking about it again I'm not sure what would be most intuitive.

MacSplody committed:
Remove file from SKSE Latest test

- used != as < would always show latest version message.

MacSplody committed:
Revert "Remove file from SKSE Latest test"

This reverts commit 8ad44be55f938172c169d97bf37e8bd7a620ffa4.

MacSplody committed:
test

MacSplody committed:
Revert "test"

This reverts commit abbd7e5d48d6167a860d257c3a51d3d818c4cbc8.

Ortham commented:
Thanks 👍

MacSplody commented:
#48 Completed
